internal/pkg/template/diff/diff.go

```diff
@@ -96,15 +98,12 @@ func (from From) ParseWithCFNIgnorer(to []byte) (Tree, error) {
 			},
 		},
 	}
-	return from.parseRoot(to, ignorer)
+	intrinsic := &intrinsicFuncFullShortFormConverter{}
+	return []overrider{ignorer, intrinsic}
 }
 
 // Parse constructs a diff tree that represent the differences of a YAML document against the From document.
-func (from From) Parse(to []byte) (Tree, error) {
-	return from.parseRoot(to, &noopOverrider{})
-}
-
-func (from From) parseRoot(to []byte, overrider overrider) (Tree, error) {
+func (from From) Parse(to []byte, overriders ...overrider) (Tree, error) {
```

Comment:
This is so interesting, I'm just thinking out loud but we are accepting a _private_ type as input to a public method.

The downside is that clients of this package can't provide their own `overrider` but can use ones that are defined in this package. I think that's okay, it's just unusual to see.

Reply:
It's totally okay to make `overrider` a public interface - if `template/diff` was a public package consumed by arbitrary clients, I would make `override` public`.

It is private in this case because we know who the clients are, and we know that they would not intend to implement their own `overrider`.

In terms of package interface design though, I think it indeed makes more sense to make it public than private.

Reply:
In fact, since 
> we know who the clients are, and we know that they would not intend to implement their own overrider.

In fact, I think perhaps I should revert to the previous interface design
```
Parse()
ParseWithCFNOverriders()
```

Reply:
done!